pkg/kn/traffic/compute.go

```diff
@@ -197,53 +197,55 @@ func errorOverWritingTag(tag string) error {
 		"add flag '--untag %s' in command to untag it", tag)
 }
 
-func errorRepeatingLatestRevision(forFlag string) error {
+func errorRepeatingRevision(forFlag string, name string) error {
 	return fmt.Errorf("repetition of identifier %s "+
-		"is not allowed, use only once with %s flag", latestRevisionRef, forFlag)
+		"is not allowed, use only once with %s flag", name, forFlag)
 }
 
 // verifies if user has repeated @latest field in --tag or --traffic flags
 // verifyInputSanity checks:
 // - if user has repeated @latest field in --tag or --traffic flags
 // - if provided traffic portion are integers
 func verifyInputSanity(trafficFlags *flags.Traffic) error {
-	var latestRevisionTag = false
-	var latestRevisionTraffic = false
 	var sum = 0
 
-	for _, each := range trafficFlags.RevisionsTags {
+	// To heck if there are duplicated @latest in --tag flags
+	revisionMap := make(map[string]int)
+	for i, each := range trafficFlags.RevisionsTags {
 		revision, _, err := splitByEqualSign(each)
 		if err != nil {
 			return err
 		}
 
-		if latestRevisionTag && revision == latestRevisionRef {
-			return errorRepeatingLatestRevision("--tag")
-
-		}
-
-		if revision == latestRevisionRef {
-			latestRevisionTag = true
+		fmt.Println("---", revision)
+		if _, exist := revisionMap[revision]; exist && revision == latestRevisionRef {
```

Comment:
We can't guarantee that revision provided in `--traffic` flag is a *revision*, because it can be a tag as well.

This check needs to be done checking whether the revision provided for --traffic flag is tag OR revision.

Reply:
But then again, even if it is a tag, we shouldn't allow it.
So this makes sense to me.

Reply:
this check is not needed as if its `@latest` that will also be in revisionMap

```
&& revision == latestRevisionRef
```